dashboard/modules/runtime_env/runtime_env_agent.py

```diff
@@ -195,25 +195,32 @@ def run_setup_with_logger():
                 serialized_runtime_env_context=serialized_context)
 
     async def DeleteURIs(self, request, context):
-        logger.info(f"Got request to delete URIS: {request.uris}.")
-
-        # Only a single URI is currently supported.
-        assert len(request.uris) == 1
-
-        uri = request.uris[0]
-
-        # Invalidate the env cache for any environments that contain this URI.
-        for env in self._working_dir_uri_to_envs.get(uri, []):
-            if env in self._env_cache:
-                del self._env_cache[env]
-
-        if self._working_dir_manager.delete_uri(uri):
-            return runtime_env_agent_pb2.DeleteURIsReply(
-                status=agent_manager_pb2.AGENT_RPC_STATUS_OK)
-        else:
-            return runtime_env_agent_pb2.DeleteURIsReply(
-                status=agent_manager_pb2.AGENT_RPC_STATUS_FAILED,
-                error_message=f"Local file for URI {uri} not found.")
+        logger.info(f"Got request to delete URIs: {request.uris}.")
+
+        failed_uris = []  # URIs that we failed to delete.
+
+        for plugin_uri in request.uris:
+            # Invalidate the env cache for any envs that contain this URI.
+            for env in self._uris_to_envs.get(plugin_uri, []):
+                if env in self._env_cache:
+                    del self._env_cache[env]
+
+            plugin, uri = _decode_plugin_uri(plugin_uri)
+            if plugin == "working_dir":
+                if not self._working_dir_manager.delete_uri(uri):
+                    failed_uris.append(uri)
+            elif plugin == "py_modules":
+                if not self._py_modules_manager.delete_uri(uri):
+                    failed_uris.append(uri)
+
```

Comment:
in general it's nice to have catch-all `else clauses with an assertion to catch current/future errors (e.g., if someone tries to add a new plugin but doesn't handle this!)

Reply:
Good idea, I'll add it in the next PR with conda GC.  When the plugins API is done this part will probably be replaced by something that programmatically goes through all the plugins (so that a plugin author doesn't have to edit this part of the code)